### PR TITLE
[Main Course] - Selection for Minimum Sample n

### PR DESCRIPTION
**Pull request by EchoTang04**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Tina0330 — APPROVED** [review]
LGTM, thanks tangtang!

File: docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+## 1 前言
+读曼昆的经济学原理的时候，印象最深刻的一句话是：People face trade-offs, the cost of something is what you give up to get it. 简单来说就是事物都有成本，每个选择都面临取舍。这句话在生产环境中尤为合适。当研发新药的时候，当金融机构对策略进行分析的时候，当判断系统某个新功能是否有效的时候，在一切需要假设检验的地方，就会面临检验样本量的选择。样本量太小，则检验的可信度不高，没有说服力；样本量太大，则成本过高，不可取。在这种情况下，如何选择使样本有代表性、使检验有效的最小样本量变得尤其重要。
+
+
+## 2 过程介绍
+
+开始之前先回顾一下假设检验的基础概念。类似于反证法，通常我们将想要拒绝的假设（比如实验组与对照组无差异）设为原假设（H0)，想要证明的假设（双边：实验组与对照组显著不同；单边：实验组明显优于/劣于对照组）设为备择假设（H1）。实验目的为推翻原假设。
+
+犯第一类错误（拒真）的概率为α，也叫做显著性水平；第二类错误（受伪）的概率为β。则power of effect检验功效为1-β，也就是当原假设为假时，正确拒绝原假设的概率。
```
Comment (BulletTech2021) [inline docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md:18]: 检验功效（power of effect）

**Review by jhqian0909 — APPROVED** [review]
LGTM. 我理解就是根据不同的问题及假设和alpha, beta能用公式确定最小样本量。
找到一个好像还不错的网站https://www.evanmiller.org/ab-testing/sample-size.html

**Comment by EchoTang04** [comment]
> LGTM. 我理解就是根据不同的问题及假设和alpha, beta能用公式确定最小样本量。 找到一个好像还不错的网站https://www.evanmiller.org/ab-testing/sample-size.html

是的，以及效应量。还有就是要判断适用什么检验。这个网站蛮好用的，可以加进reference里。感谢佳辉！

File: docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md (hunk 1 above)
Comment (EchoTang04) [reply docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md]: > 检验功效（power of effect）

已修改~

File: docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+- 单样本还是两样本？（影响标准差的计算）
+- 单边检验还是双边检验？（影响查表统计量分位数α或α/2的选择）
+- 等等。
+
+考虑到方差的检验可以另开一篇方差分析来讲（包括F检验，ANOVA等），这里我们侧重均值/比率的假设检验。
+
+### 2.2 确定检验方法
+
+明确好研究对象之后，我们就可以选择合适的检验方法。若对象为均值，多采用Z检验/t检验；若为比率，则可以考虑采用卡方检验。当确定了检验方法，我们可以写出统计量的公式，反推之后即可得到最小样本量n的计算公式。统计量公式繁多，这里不再赘述。均值公式举例如下图。注意，如为双边检验，则将α改成α/2即可。
+<figure>
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md:40]: 不同格式的地方最好空一行

File: docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#设定显著性水平为0.05，检验功效为0.8，效应量为标准均差计算方式，对照组和实验组的（预期）比率分别为30%和33%，实验组和对照组数量相等，双边检验。
+GofChisquarePower().solve_power(effect_size=es(prop1=0.33, prop2=0.30, method= 'normal'), alpha=0.05, power=0.8,n_bins=2)
+
+#z检验
+#设定显著性水平为0.05，检验功效为0.8，效应量为标准均差计算方式，对照组和实验组的（预期）比率分别为30%和33%，实验组和对照组数量相等，双边检验。
+zt_ind_solve_power(effect_size=es(prop1=0.33, prop2=0.30, method= 'normal'), alpha=0.05, power=0.8, ratio=1.0, alternative="two-sided")
+```
+结果如下。
+
+<figure>
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md:70]: 结果可以只用用文字写出来，因为图片在手机上看起来很小

**Review by Realvincentyuan — APPROVED** [review]
梦回研究生检验实验结果的时候，谢谢Echo的分享！

File: docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+</figure>
+
+注意此处为过剩近似，且卡方检验显示的最小样本量为每组样本量，z检验显示的最小样本量为试验的总样本量。可知每组最小样本量为1881，总样本量为3762。
+
+此外：常用α为0.05，常用β为0.2。从中我们也可以看出偏好。人们倾向于严格控制拒真的概率，为此宁愿付出接受错误原假设的代价。
+
+## 3 小结
+假设检验方法千千万，但万变不离其宗。只要明确了研究对象，接下来的步骤就顺理成章非常顺滑。样本大小、显著性水平α、检验功效（1-β）、效应量，给定其中任意三个量，我们就可以推算出第四个量。
+
+最后留一个开放性问题。还是刚刚的例子，当期待B型广告的点击率提升的更多时，需要的最小样本量是增多还是减少呢？
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md:82]: 提升`的` -》 提升`得`

**Review by Realvincentyuan — APPROVED** [review]
`的`和`得`的选择可以修改下

File: docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md (hunk 2 above)
Comment (EchoTang04) [reply docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md:40]: > 不同格式的地方最好空一行

已修改~感谢提醒~

File: docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md (hunk 3 above)
Comment (EchoTang04) [reply docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md:70]: 嗯嗯，文字描述在2.3里也写了。

File: docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md (hunk 4 above)
Comment (EchoTang04) [reply docs/Main_Course/Data_Analytics/2021-12-15-min_n_choice.md:82]: > 提升`的` -》 提升`得`

哈哈哈感谢提醒！这个的和得我从小就分不清楚

**Pull request merged**